### PR TITLE
Change: Use SlErrorCorrupt() on pool index error when loading a savegame, instead of terminating.

### PR DESCRIPTION
**Pull request by PeterN**

### CODE REVIEW

**Comment by PeterN** [comment]
Hmm, I may rework this to be just one function that takes varargs.

File: src/core/pool_func.cpp [hunk 1]
```diff
@@ -38,3 +42,19 @@
...
+	char msg[256];
+
+	va_start(ap, format);
+	vseprintf(msg, lastof(msg), format, ap);
+	va_end(ap);
+
+	if (_switch_mode == SM_LOAD_GAME || _switch_mode == SM_LOAD_SCENARIO) {
+		SlErrorCorrupt(msg);
+	} else {
+		usererror("Failed loading savegame: %s", msg);
```
Comment (LordAro) [inline src/core/pool_func.cpp]: this confuses me - you have a check that you're currently loading a game/scenario, but then this other message includes "loading savegame"

They can't both be correct?

File: src/core/pool_func.cpp [hunk 2]
```diff
@@ -12,6 +12,10 @@
 #include "../stdafx.h"
 #include "pool_type.hpp"
 
+#include "../openttd.h"
+#include "../string_func.h"
+#include "../saveload/saveload.h"
```
Comment (LordAro) [inline src/core/pool_func.cpp]: fair bit of overhead adding 3 header includes. Perhaps some forward declarations might be easier on the compiler?

**Review by LordAro — CHANGES_REQUESTED** [review]

File: src/core/pool_func.cpp (hunk 1 above)
Comment (PeterN) [reply src/core/pool_func.cpp]: Er... good point o_O. At the point when I was checking we are loading a save that text had got lost from the usererror, but was put back later. Hmm!

**Comment by PeterN** [comment]
Iteration 4... No special PoolBase::Error() function now, just a new SlErrorCorruptFmt() that takes a format string.

**Review by LordAro — APPROVED** [review]

**Pull request merged**